docs/developer-docs/backend/rust/10-timers.md

```diff
@@ -0,0 +1,237 @@
+- `ic_cdk::println!(...)` prints the debug log message on the local `dfx` console.
+- `ic_cdk_timers::set_timer_interval(...)` creates a new periodic timer with the specified interval and a closure to call.
+- `COUNTER.fetch_add(1, ...)` increases the global `COUNTER` every time the periodic task is triggered.
+
+## Implementing canister upgrade
+
+:::info
+Note: As described in the [periodic tasks and timers](../periodic-tasks) page, the timers library does not handle canister upgrades. It is up to the canister developer to serialize the timers in the `canister_pre_upgrade` and reactivate the timers in the `canister_post_upgrade` method if needed.
+:::
+
+For the sake of simplicity, in this guide the `canister_post_upgrade` method just calls `canister_init` to reinitialize the timer.
+
+In the code editor, open the `src/my_timers_backend/src/lib.rs` file and append the following:
+
+```rust
+[...]
+
+#[ic_cdk::post_upgrade]
+fn post_upgrade(timer_interval_secs: u64) {
+    init(timer_interval_secs)
+}
+```
+
+This code establishes the following:
+
+- `#[ic_cdk::post_upgrade]` marks the following `post_upgrade` function as a canister post-upgrade handler, so the function will be exported as `canister_post_upgrade`.
+- `fn post_upgrade(interval: u64) {...}` defines the post-upgrade method. Just like in the `.did` definition, the function takes one argument: timer interval in seconds.
+- `init(timer_interval_secs)` for the sake of simplicity, the post-upgrade just calls the `init` function, i.e. does exactly the same as the canister initialization.
+
+The canister's code is now complete. The finished file should look like this:
+
+`src/my_timers_backend/src/lib.rs`:
+
+```rust
+use std::sync::atomic::{AtomicU64, Ordering};
+
+static COUNTER: AtomicU64 = AtomicU64::new(0);
+
+#[ic_cdk::query]
+fn counter() -> u64 {
+    COUNTER.load(Ordering::Relaxed)
+}
+
+#[ic_cdk::init]
+fn init(timer_interval_secs: u64) {
+    let interval = std::time::Duration::from_secs(timer_interval_secs);
+    ic_cdk::println!("Starting a periodic task with interval {interval:?}");
+    ic_cdk_timers::set_timer_interval(interval, || {
+        COUNTER.fetch_add(1, Ordering::Relaxed);
+    });
+}
+
+#[ic_cdk::post_upgrade]
+fn post_upgrade(timer_interval_secs: u64) {
+    init(timer_interval_secs)
+}
+```
+
+Save the file.
+
+## Running the dapp locally
+
+The libraries are added, the canister interface is described and the code is complete. Time to try it all out!
+
+Start by assuring that you are still in your project's directory. Then, start the local execution environment with the command:
+
+```sh
+dfx start --clean --background
+```
+
+Then, compile and deploy `my_timers_backend` canister, setting the interval for the periodic task to 1s:
+
+```sh
+dfx deploy my_timers_backend --argument 1
+```
+
+The counter inside the canister starts increasing every second.
+
+Example output:
+
+```sh
+% dfx deploy my_timers_backend --argument 1
```

Comment:
```suggestion
dfx deploy my_timers_backend --argument 1
```
We want to be consistent about this (and if we later decide instead that we do want to use a character, it should be `$`).